packages/js/product-editor/src/hooks/use-product-attributes.ts

```diff
@@ -78,27 +77,72 @@ export function useProductAttributes( {
 		};
 	};
 
-	const getAugmentedAttributes = ( atts: ProductAttribute[] ) => {
+	const getAugmentedAttributes = (
+		atts: ProductAttribute[],
+		variation: boolean,
+		startPosition: number
+	) => {
 		return atts.map( ( attribute, index ) => ( {
 			...attribute,
-			variation: isVariationAttributes,
-			position: attributes.length + index,
+			variation,
+			position: startPosition + index,
 		} ) );
 	};
 
 	const handleChange = ( newAttributes: ProductAttribute[] ) => {
-		const augmentedAttributes = getAugmentedAttributes( newAttributes );
-		const otherAttributes = isVariationAttributes
+		let otherAttributes = isVariationAttributes
 			? allAttributes.filter( ( attribute ) => ! attribute.variation )
 			: allAttributes.filter( ( attribute ) => !! attribute.variation );
-		setAttributes( augmentedAttributes );
-		onChange( [ ...otherAttributes, ...augmentedAttributes ] );
+
+		// Remove duplicate global attributes.
+		otherAttributes = otherAttributes.filter( ( attr ) => {
+			if (
+				attr.id > 0 &&
+				newAttributes.findIndex( ( a ) => a.id === attr.id ) !== -1
```

Comment:
You can use `some` function ti check if at least one item exists within an array. 

```suggestion
				newAttributes.some( ( a ) => a.id === attr.id )
```

Reply:
Done as part of https://github.com/woocommerce/woocommerce/pull/39522/commits/f8f2b9ef2f5cec258961ddfea82c87595c93d305